### PR TITLE
Avoid unneeded warning when decoding invalid data on PHP 7.4

### PR DESCRIPTION
**Pull request by WyriHaximus**
Supersedes / Closes #356
Refs: https://github.com/reactphp/http-client/pull/150

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,35 +1,30 @@
...
 
 install:
   - composer install --no-interaction
   - if [ "$DEPENDENCIES" = "lowest" ]; then composer update --prefer-lowest -n; fi
-  
+
 script:
   - ./vendor/bin/phpunit --coverage-text
-  - if [ "$DEPENDENCIES" = "lowest" ]; then php -n tests/benchmark-middleware-runner.php; fi
+  - if [ "$DEPENDENCIES" = "lowest" ]; then php -n tests/benchmark-middleware-runner.php; fi
```
Comment (clue) [inline .travis.yml]: nit: eol

File: .travis.yml [hunk 2]
```diff
@@ -1,35 +1,30 @@
...
 matrix:
   include:
     - php: 5.3
       dist: precise
+    - php: 5.4
+    - php: 5.5
+    - php: 5.6
     - php: 7.0
-      env:
-        - DEPENDENCIES=lowest
```
Comment (clue) [inline .travis.yml]: Intentionally removed? Not sure how much value this still provides, but the environment variable is still referenced below. Perhaps keep this and discuss this in a separate ticket?

File: .travis.yml (hunk 1 above)
Comment (WyriHaximus) [reply .travis.yml]: :man_facepalming: Lazy copy pasta xD

File: .travis.yml (hunk 2 above)
Comment (WyriHaximus) [reply .travis.yml]: Not intentionally removed, added it again

File: .travis.yml [hunk 3]
```diff
@@ -1,35 +1,33 @@
...
-php:
-# - 5.3 # requires old distro, see below
-  - 5.4
-  - 5.5
-  - 5.6
-  - 7.0
-  - 7.1
-  - 7.2
-  - hhvm # ignore errors, see below
-
```
Comment (clue) [inline .travis.yml]: Nit: Unrelated change, but I don't want to hold this back because of this :+1:

Comment (WyriHaximus) [reply .travis.yml]: Dropped the `-VER`

Comment (clue) [reply .travis.yml]: I meant to refer to the whole `php:` vs `matrix:` block, but it's alright as-is IMHO :+1:

**Pull request merged**